fv3config/config/diag_table.py

```diff
@@ -0,0 +1,212 @@
+import logging
+from typing import Sequence, Optional, Union
+import dataclasses
+import datetime
+import re
+
+import dacite
+
+from .._exceptions import ConfigError
+
+logger = logging.getLogger("fv3config")
+
+
+@dataclasses.dataclass
+class DiagTableField:
+    """Object representing a field of a diagnostics file."""
+
+    module_name: str
+    field_name: str
+    output_name: str
+    reduction_method: Optional[str] = "none"
```

Comment:
with Python 3.8, could use `Literal["none", "average", "min", "max"]` as type hint, which would make for awesome self-documentation and runtime validation.

Reply:
This could also be an Enum class.

Reply:
We can't upgrade to 3.8. All the fv3 docker images we use are based on ubuntu:18.04 which has python 3.6.

Reply:
Let's avoid upgrading for now but put this in the box of minor reasons to upgrade our ubuntu version.